algebird-test/src/test/scala/com/twitter/algebird/BloomFilterTest.scala

```diff
@@ -376,4 +400,56 @@ class BloomFilterTest extends WordSpec with Matchers {
       }
     }
   }
+
+  "BloomFilters" should {
+    "should not be able to compute Hamming distance to each other when they are not " +
+      "of equal width and have the same number of hashes" in {
+
+        val elems = Seq("A", "B", "C")
+
+        val bfMonoid1 = new BloomFilterMonoid[String](numHashes = 4, width = 64)
+        val bf1 = bfMonoid1.create(elems: _*)
+
+        val bfMonoid2 = new BloomFilterMonoid[String](numHashes = 5, width = 64)
+        val bf2 = bfMonoid2.create(elems: _*)
+
+        val bfMonoid3 = new BloomFilterMonoid[String](numHashes = 5, width = 128)
+        val bf3 = bfMonoid3.create(elems: _*)
+
+        assertThrows[AssertionError]{
+          BloomFilter.hammingDistance(bf1, bf2)
+        }
+
+        assertThrows[AssertionError]{
+          BloomFilter.hammingDistance(bf2, bf3)
+        }
+
+      }
+
+    "be able to compute compute Hamming distance to each other" in {
+      import BloomFilterTestUtils._
+
+      def createBFWithItems(entries: Seq[String]): BF[String] = {
+        val numOfHashes = 3
+        val width = 64
+        val bfMonoid = new BloomFilterMonoid[String](numOfHashes, width)
+        bfMonoid.create(entries: _*)
+      }
+
+      val firstBloomFilter = createBFWithItems(Seq("A"))
+      val secondBloomFilter = createBFWithItems(Seq("C"))
+
+      val distance1 = BloomFilter.hammingDistance(firstBloomFilter, secondBloomFilter)
+      assert(distance1 === 4)
+
+      val thirdBloomFilter = createBFWithItems(Seq("A", "B", "C"))
+      // Make it dense to make sure that that case is also covered
+      // even though these examples are small and thus sparse.
+      val forthBloomFilter = toDense(createBFWithItems(Seq("C", "D", "E")))
+
```

Comment:
can you add the `createBFWithItems(List())` case as well, just to test emptiness again?